hal/src/samd51/qspi.rs

```diff
@@ -0,0 +1,219 @@
+use crate::{
+    target_device::{QSPI, MCLK}, 
+    gpio::{Pa8, Pa9, Pa10, Pa11, Pb10, Pb11, Input, Floating, PfH, Port},
+};
+
+pub struct Qspi {
+    qspi: QSPI,
+    sck: Pb10<PfH>,
+    cs:  Pb11<PfH>,
+    io0: Pa8<PfH>,
+    io1: Pa9<PfH>,
+    io2: Pa10<PfH>,
+    io3: Pa11<PfH>,
+}
+
+impl Qspi {
+   pub fn new(
+        mclk: &mut MCLK,
+        port: &mut Port,
+        qspi: QSPI,
+        sck: Pb10<Input<Floating>>,
+        cs:  Pb11<Input<Floating>>,
+        io0: Pa8<Input<Floating>>,
+        io1: Pa9<Input<Floating>>,
+        io2: Pa10<Input<Floating>>,
+        io3: Pa11<Input<Floating>>,
+    ) -> Qspi {
+        let sck = sck.into_function_h(port);
+        let cs = cs.into_function_h(port);
+        let io0 = io0.into_function_h(port); 
+        let io1 = io1.into_function_h(port); 
+        let io2 = io2.into_function_h(port); 
+        let io3 = io3.into_function_h(port); 
+
+        mclk.apbcmask.modify(|_, w| w.qspi_().set_bit());
+        mclk.ahbmask.modify(|_, w| w.qspi_().set_bit());
```

Comment:
Could you roll the two`ahbmask` writes into one call to `modify()`? Also maybe add a comment, ie: _Enable the clocks for the qspi peripheral in single data rate mode._